### PR TITLE
New blog post on our Carbon Hack participation

### PR DESCRIPTION
**Pull request by mgriffin-scottlogic**
Please add a direct link to your post here:

https://mgriffin-scottlogic.github.io/blog/2024/04/11/Carbon-Hack-24-Leveraging-the-Impact-Framework-to-Estimate-the-Carbon-Cost-of-Cloud-Storage.html

Have you (please tick each box to show completion):

 - [x] Added your blog post to a single category?
 - [x] Added a brief summary for your post? Summaries should be roughly two sentences in length and give potential readers a good idea of the contents of your post.
 - [x] Checked that the build passes?
 - [x] Checked your spelling (you can use `npm spellcheck` if that's your thing)
 - [x] Ensured that your author profile contains a profile image, and a brief description of yourself? (make it more interesting than just your job title!)
 - [x] Optimised any images in your post? They should be less than 100KBytes as a general guide.

Posts are reviewed / approved by your Regional Tech Lead.

@taddison-scottlogic please review / approve if possible.

### CODE REVIEW

File: _posts/2024-04-11-Carbon-Hack-24--Leveraging-the-Impact Framework-to-Estimate-the-Carbon-Cost-of-Cloud-Storage.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+Once we had this setup, we first created a simple plugin, which we named ‘[Ramboll-Resilio](https://github.com/mgriffin-scottlogic/if-carbon-hack-plugin/tree/main/src/lib/ramboll-resilio)’ based on the authors of a paper called [Assessment of the energy footprint of digital actions and services](https://op.europa.eu/en/publication-detail/-/publication/d3b6c0a1-1171-11ee-b12e-01aa75ed71a1). Part of this paper proposes a simple figure for calculating the energy used to store a Gb of data for a year, which the plugin implements. The paper itself notes that this uses some arbitrary figures and relies on Googles own data, which isn’t ideal but can be a starting point in trying to make an attributional estimation.
```
Comment (taddison-scottlogic) [inline _posts/2024-04-11-Carbon-Hack-24--Leveraging-the-Impact Framework-to-Estimate-the-Carbon-Cost-of-Cloud-Storage.md]: "to store a Gb of data" - can I just check whether you mean gigabit or gigabyte? Should this be GB (sorry - I'm being pedantic!)

File: _posts/2024-04-11-Carbon-Hack-24--Leveraging-the-Impact Framework-to-Estimate-the-Carbon-Cost-of-Cloud-Storage.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Another inspiration that the Cloud Carbon Footprint tool provided was their approach to energy consumed by [storage](https://www.cloudcarbonfootprint.org/docs/methodology#storage). Within the tool it uses two different coefficients depending on whether data is stored on HDD or SSD, but this is tied to a point in time industry average in terms of disk capacity and power consumption. An Impact Framework plugin could rely on these figures being passed to it instead, allowing the choice to use an average or specific drive information if known. To perform this calculation we added the [storage-energy](https://github.com/mgriffin-scottlogic/if-carbon-hack-plugin/tree/main/src/lib/storage-energy) plugin.
+
+To complete the storage focus we also added a [storage-read-write-energy](https://github.com/mgriffin-scottlogic/if-carbon-hack-plugin/tree/main/src/lib/storage-read-write-energy) plugin, which requires the power demand of a drive when reading or writing (which can be significantly different than when idle) and its read/write speed in Gb/s. This is used to calculate the minimum time that it must take to transfer data and the drive energy that would be consumed in the process.
```
Comment (taddison-scottlogic) [inline _posts/2024-04-11-Carbon-Hack-24--Leveraging-the-Impact Framework-to-Estimate-the-Carbon-Cost-of-Cloud-Storage.md]: "and its read/write speed in Gb/s" - I suspect this one is meant to be Gb/s, but just checking.

**Review by taddison-scottlogic — COMMENTED** [review]
Looks good and reads nicely to me. I don't suppose there's any imagery that could be added to bring things more to life?

File: _posts/2024-04-11-Carbon-Hack-24--Leveraging-the-Impact Framework-to-Estimate-the-Carbon-Cost-of-Cloud-Storage.md (hunk 1 above)
Comment (mgriffin-scottlogic) [reply _posts/2024-04-11-Carbon-Hack-24--Leveraging-the-Impact Framework-to-Estimate-the-Carbon-Cost-of-Cloud-Storage.md]: No worries, I forget about the distinction, I think everything should be talking in terms of Gigabytes so should be capital GB

File: _posts/2024-04-11-Carbon-Hack-24--Leveraging-the-Impact Framework-to-Estimate-the-Carbon-Cost-of-Cloud-Storage.md (hunk 2 above)
Comment (mgriffin-scottlogic) [reply _posts/2024-04-11-Carbon-Hack-24--Leveraging-the-Impact Framework-to-Estimate-the-Carbon-Cost-of-Cloud-Storage.md]: Again I think the code is written to work with Gigabytes so I'll clarify

**Comment by mgriffin-scottlogic** [comment]
> Looks good and reads nicely to me. I don't suppose there's any imagery that could be added to bring things more to life?

There isn't too much in terms of imagery, unless we include the cover sheet from our submission video?

Carbon Hack 24 - Project Submission

**Pull request merged**